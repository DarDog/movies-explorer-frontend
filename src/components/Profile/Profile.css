.profile {
  padding: 75px 0 70px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
}

.profile__title {
  margin: 0 0 120px;
  padding: 0;
  font-weight: 500;
  font-size: 24px;
  line-height: 29px;
  text-align: center;
  color: black;
}

.profile__form {
  min-width: 400px;
  display: flex;
  flex-direction: column;
  margin-bottom: 15px;
}

.form__element {
  display: flex;
  justify-content: space-between;
  margin-bottom: 15px;
}

.form__element:first-of-type {
  padding-bottom: 15px;
  border-bottom: 1px solid #E8E8E8;
}

.form__label {
  font-weight: 500;
  font-size: 11px;
  line-height: 13px;
  color: black;
}

.form__input {
  border: none;
  padding: 0;
  text-align: right;
  font-weight: 400;
  font-size: 11px;
  line-height: 13px;
  color: black;
  outline: none;
}

.form__input_error {

}

.form__button {
  border: none;
  outline: none;
  background-color: inherit;
  font-weight: 400;
  font-size: 13px;
  line-height: 16px;
  color: black;
  margin-top: 220px;
  transition: .3s;
  cursor: pointer;
}

.form__button:hover {
  transform: scale(1.1);
  opacity: .8;
}

.form__button_disable {
  cursor: auto;
  color: #a0a0a0;
  pointer-events: none;
}

.form__button_disable:hover {
  transform: scale(1);
  opacity: 1;
}

.profile__exit {
  border: none;
  outline: none;
  background-color: inherit;
  font-weight: 500;
  font-size: 13px;
  line-height: 16px;
  color: #EE3465;
  transition: .3s;
  cursor: pointer;
}

.profile__exit:hover {
  transform: scale(1.1);
  opacity: .8;
}

.form__error {
  font-size: 10px;
  line-height: 12px;
  color: #EE3465;
  margin-bottom: 15px;
}

.form__error_position_center {
  align-self: center;
  text-align: center;
}

.form__success {
  font-size: 10px;
  line-height: 12px;
  color: #2be080;
  margin-bottom: 15px;
  text-align: center;
}


@media screen and (max-width: 500px){
  .profile {
    padding: 70px 30px 40px;
  }

  .profile__title {
    margin-bottom: 80px;
  }
  .profile__form {
    min-width: 260px;
  }

  .form__button {
    font-size: 12px;
    line-height: 15px;
  }
}
