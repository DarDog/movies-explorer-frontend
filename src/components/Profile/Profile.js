import React, { useEffect } from 'react';
import Header from "../Header/Header";
import './Profile.css'
import { useAuth } from "../../hooks/useAuth";
import { useNavigate } from "react-router-dom";
import { useForm } from "../../hooks/useForm";

const Profile = (props) => {
  const { user, signOut, setUserInfo } = useAuth();
  const navigate = useNavigate();
  const { handleChange, values, errors, isValid, setValues, setIsValid } = useForm();
  let errCode;

  useEffect(() => {
    setValues({name: user.name, email: user.email})
  }, []);

  useEffect(() => {
    if (values.name === user.name && values.email === user.email) {
      setIsValid(false);
    }
  }, [values])

  const handleSubmit = (evt) => {
    evt.preventDefault();
    if (isValid) {
      setUserInfo({ email: values.email, name: values.name })
    }
  }

  const handleSignOut = () => {
    signOut(() => navigate('/', { replace: true }))
  }

  return (
    <>
      <Header handleClick={ props.handlePopupOpen }/>
      <section className='profile'>
        <h1 className='profile__title'>Привет, { user.name }!</h1>
        <form onSubmit={ handleSubmit } name='edit-profile' className='profile__form' noValidate>
          <div className='form__element'>
            <label className='form__label' htmlFor='name'>Имя</label>
            <input className='form__input' name='name' type='text' placeholder={ user.name } value={ values.name || '' }
                   onChange={ handleChange }
                   minLength='2' maxLength='30' required/>
          </div>
          <span className='form__error'>{ errors.name }</span>
          <div className='form__element'>
            <label className='form__label' htmlFor='email'>E-mail</label>
            <input className='form__input' name='email' type='email' placeholder={ user.email }
                   value={ values.email || '' } onChange={ handleChange }
                   required/>
          </div>
          <span className='form__error'>{ errors.email }</span>
          <button type='submit'
                  className={ `form__button ${ isValid ? '' : 'form__button_disable' }` }>Редактировать
          </button>
        </form>
        <button type='button' className='profile__exit' onClick={ handleSignOut }>Выйти из аккаунта</button>
      </section>
    </>
  );
}

export default Profile
