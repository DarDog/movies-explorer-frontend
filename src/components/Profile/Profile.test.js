import React from 'react';
import Profile from "./Profile";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import '@testing-library/jest-dom';

describe('Компонент Profile', () => {
  it('Компонент рендерится', () => {
    render(
      <BrowserRouter>
        <Profile/>
      </BrowserRouter>
    );

    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(screen.getByRole('form')).toBeInTheDocument();
    expect(screen.getAllByRole('textbox')).toHaveLength(2);
    expect(screen.getAllByRole('button')).toHaveLength(2);
  })
  it('Соответствует своему snapshot', () => {
    const component = render(
      <BrowserRouter>
        <Profile/>
      </BrowserRouter>
    );

    expect(component).toMatchSnapshot();
  })
})
