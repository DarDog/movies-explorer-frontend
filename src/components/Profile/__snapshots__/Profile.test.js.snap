// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Profile Соответствует своему snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header page__header"
      >
        <nav
          class="navigation"
        >
          <a
            aria-current="page"
            class="logo active"
            href="/"
          />
          <a
            class="navigation__link"
            href="/movies"
          >
            Фильмы
          </a>
          <a
            class="navigation__link"
            href="/saved-movies"
          >
            Сохранённые фильмы
          </a>
        </nav>
        <a
          class="header__account-link"
          href="/profile"
        >
          Аккаунт
        </a>
      </header>
      <section
        class="profile"
      >
        <h1
          class="profile__title"
        >
          Привет, Владислав!
        </h1>
        <form
          class="profile__form"
          name="edit-profile"
        >
          <div
            class="form__element"
          >
            <label
              class="form__label"
            >
              Имя
            </label>
            <input
              class="form__input"
              placeholder="Владислав"
              type="text"
            />
          </div>
          <div
            class="form__element"
          >
            <label
              class="form__label"
            >
              E-mail
            </label>
            <input
              class="form__input"
              placeholder="mail@mail.com"
              type="email"
            />
          </div>
          <button
            class="form__button"
            type="submit"
          >
            Редактировать
          </button>
        </form>
        <button
          class="profile__exit"
          type="button"
        >
          Выйти из аккаунта
        </button>
      </section>
    </div>
  </body>,
  "container": <div>
    <header
      class="header page__header"
    >
      <nav
        class="navigation"
      >
        <a
          aria-current="page"
          class="logo active"
          href="/"
        />
        <a
          class="navigation__link"
          href="/movies"
        >
          Фильмы
        </a>
        <a
          class="navigation__link"
          href="/saved-movies"
        >
          Сохранённые фильмы
        </a>
      </nav>
      <a
        class="header__account-link"
        href="/profile"
      >
        Аккаунт
      </a>
    </header>
    <section
      class="profile"
    >
      <h1
        class="profile__title"
      >
        Привет, Владислав!
      </h1>
      <form
        class="profile__form"
        name="edit-profile"
      >
        <div
          class="form__element"
        >
          <label
            class="form__label"
          >
            Имя
          </label>
          <input
            class="form__input"
            placeholder="Владислав"
            type="text"
          />
        </div>
        <div
          class="form__element"
        >
          <label
            class="form__label"
          >
            E-mail
          </label>
          <input
            class="form__input"
            placeholder="mail@mail.com"
            type="email"
          />
        </div>
        <button
          class="form__button"
          type="submit"
        >
          Редактировать
        </button>
      </form>
      <button
        class="profile__exit"
        type="button"
      >
        Выйти из аккаунта
      </button>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
